src/fheroes2/dialog/dialog_file.cpp

```diff
@@ -97,8 +97,9 @@ fheroes2::GameMode Dialog::FileOptions()
                 fheroes2::showMessage( fheroes2::Text( _( "Load Game" ), fheroes2::FontType::normalYellow() ),
                                        fheroes2::Text( _( "No save files to load." ), fheroes2::FontType::normalWhite() ), Dialog::OK );
             }
-            else {
-                result = Interface::AdventureMap::Get().EventLoadGame();
+
+            if ( Interface::AdventureMap::Get().EventLoadGame() == fheroes2::GameMode::LOAD_GAME ) {
+                result = fheroes2::GameMode::LOAD_GAME;
                 break;
             }
```

Comment:
We should not remove the `else` statement. This new code should be executed only if `ListFiles::isEmpty()` returned `false` because we should not open Load Game dialog if there are no saved games.

```suggestion
            else {
                if ( Interface::AdventureMap::Get().EventLoadGame() == fheroes2::GameMode::LOAD_GAME ) {
                    result = fheroes2::GameMode::LOAD_GAME;
                    break;
                }
            }
```

Reply:
You are right, fixed.